aptos-move/block-executor/src/view.rs

```diff
@@ -1257,6 +1346,51 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> ValueToIden
     }
 }
 
+struct TemporaryExtractIdentifiersMapping<T: Transaction> {
+    // These are the delayed field keys that were touched when utilizing this mapping
+    // to replace ids with values or values with ids
+    delayed_field_keys: RefCell<HashSet<T::Identifier>>,
+}
+
+impl<T: Transaction> TemporaryExtractIdentifiersMapping<T> {
+    pub fn new() -> Self {
+        Self {
+            delayed_field_keys: RefCell::new(HashSet::new()),
+        }
+    }
+
+    pub fn into_inner(self) -> HashSet<T::Identifier> {
+        self.delayed_field_keys.into_inner()
+    }
+}
+
+impl<T: Transaction> ValueToIdentifierMapping for TemporaryExtractIdentifiersMapping<T> {
+    fn value_to_identifier(
+        &self,
+        _kind: &IdentifierMappingKind,
+        layout: &MoveTypeLayout,
+        value: Value,
+    ) -> TransformationResult<Value> {
+        let id = T::Identifier::try_from_move_value(layout, value, &())
+            .map_err(|e| TransformationError(format!("{:?}", e)))?;
+        self.delayed_field_keys.borrow_mut().insert(id);
```

Comment:
Why? The `value` here is a real value, not an ID?

Reply:
I see now, we deserialise the value with exchange values, so here value is already an ID - would be nice to mention that in the comment?